### PR TITLE
add io.github.gopher64.gopher64

### PR DESCRIPTION
**Pull request by loganmc10**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

**Comment by loganmc10** [comment]
bot, build io.github.gopher64.gopher64

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "only-arches": [
+    "x86_64"
+  ],
+  "automerge-flathubbot-prs": true
```
Comment (hfiguiere) [inline flathub.json:5]:
```suggestion
  ]
```

Comment (loganmc10) [reply flathub.json:5]: I would prefer to have the flathubbot PRs get merged automatically

**Comment by loganmc10** [comment]
bot, build io.github.gopher64.gopher64

File: io.github.gopher64.gopher64.desktop
Comment (hfiguiere) [inline io.github.gopher64.gopher64.desktop:1]: this file should be upstream

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "only-arches": [
+    "x86_64"
```
Comment (hfiguiere) [inline flathub.json:3]: why not building on aarch64 too?

File: io.github.gopher64.gopher64.metainfo.xml
Comment (hfiguiere) [inline io.github.gopher64.gopher64.metainfo.xml:1]: this file should be upstream too

File: io.github.gopher64.gopher64.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+app-id: io.github.gopher64.gopher64
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+command: /app/bin/gopher64
```
Comment (hfiguiere) [inline io.github.gopher64.gopher64.yaml:5]:
```suggestion
command: gopher64
```

File: io.github.gopher64.gopher64.desktop
Comment (loganmc10) [reply io.github.gopher64.gopher64.desktop:1]: Is this documented somewhere?

File: io.github.gopher64.gopher64.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  - --share=ipc
+  - --socket=pulseaudio
+  - --device=all
+
+modules:
+  - name: gopher64
+    buildsystem: simple
+    build-commands:
+      - install -D gopher64 -t /app/bin
+      - chmod +x /app/bin/gopher64
```
Comment (hfiguiere) [inline io.github.gopher64.gopher64.yaml:19]:
```suggestion
      - install -Dm755 gopher64 -t /app/bin
```

File: io.github.gopher64.gopher64.metainfo.xml
Comment (loganmc10) [reply io.github.gopher64.gopher64.metainfo.xml:1]: Where is this documented?

File: flathub.json (hunk 2 above)
Comment (loganmc10) [reply flathub.json:3]: The program uses x64 assembly

File: io.github.gopher64.gopher64.desktop
Comment (hfiguiere) [reply io.github.gopher64.gopher64.desktop:1]: the 20 year standard that the .desktop file are? https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-1.0.html

about having them upstream
https://docs.flathub.org/docs/for-app-authors/requirements#acceptable-but-should-be-submitted-upstream


neither are specific to flatpak.

File: io.github.gopher64.gopher64.metainfo.xml
Comment (loganmc10) [reply io.github.gopher64.gopher64.metainfo.xml:1]: I don't fully understand this one. Doesn't the flathub bot update this file (the releases section)?

File: io.github.gopher64.gopher64.desktop
Comment (loganmc10) [reply io.github.gopher64.gopher64.desktop:1]: the desktop file is only used by the flatpak, so I'd rather leave it here

File: io.github.gopher64.gopher64.metainfo.xml
Comment (hfiguiere) [reply io.github.gopher64.gopher64.metainfo.xml:1]: never.

File: io.github.gopher64.gopher64.desktop
Comment (hfiguiere) [reply io.github.gopher64.gopher64.desktop:1]: it's for any Linux of BSD. Any GUI application that doesn't come with one, it's a bug.

File: io.github.gopher64.gopher64.metainfo.xml
Comment (loganmc10) [reply io.github.gopher64.gopher64.metainfo.xml:1]: Look at this: https://github.com/flathub/io.github.simple64.simple64/pull/66

File: io.github.gopher64.gopher64.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          version-query: .tag_name
+          url-query: .assets[] | select(.name|contains("linux")) | .browser_download_url
+
+  - name: finalize
+    buildsystem: simple
+    build-commands:
+      - install -D gopher64.svg -t /app/share/icons/hicolor/scalable/apps
+      - install -D io.github.gopher64.gopher64.desktop -t /app/share/applications
+      - install -D io.github.gopher64.gopher64.metainfo.xml -t /app/share/metainfo
+    sources:
```
Comment (hfiguiere) [inline io.github.gopher64.gopher64.yaml:37]:
```suggestion
      - install -D gopher64.svg -t /app/share/icons/hicolor/scalable/apps
      - install -D io.github.gopher64.gopher64.desktop -t /app/share/applications
      - install -D io.github.gopher64.gopher64.metainfo.xml -t /app/share/metainfo
    sources:
      - type: archive
        url: https://github.com/gopher64/gopher64/releases/download/v0.1.1/gopher64-linux.zip
        sha256: 9d0d4415e202dae05257ca2358d8950770b59af27673fbae245dac512805511e
        x-checker-data:
          is-main-source: true
          type: json
          url: https://api.github.com/repos/gopher64/gopher64/releases/latest
          version-query: .tag_name
          url-query: .assets[] | select(.name|contains("linux")) | .browser_download_url
```

File: io.github.gopher64.gopher64.metainfo.xml
Comment (hfiguiere) [reply io.github.gopher64.gopher64.metainfo.xml:1]: that's because it's not upstream like it should be.

Comment (loganmc10) [reply io.github.gopher64.gopher64.metainfo.xml:1]: I don't understand what you're saying. The flathub external data checker never touches the metadata file except when it does when the file isn't upstream like it should be?

**Comment by loganmc10** [comment]
I believe this meets all the app requirements and builds successfully. It is ready to be merged

**Pull request closed without merging**